src/signals.jl

```diff
@@ -68,6 +68,17 @@ const SamplesInfo = @row("onda.samples-info@1",
                          sample_type::AbstractString = onda_sample_type_from_julia_type(sample_type),
                          sample_rate::LPCM_SAMPLE_TYPE_UNION = convert_number_to_lpcm_sample_type(sample_rate))
 
+
+"""
+    Base.copy(s::SamplesInfo)
+
+Create a new `SamplesInfo` by performing a shallow `copy` of each of its fields, if possible according to `Base.applicable`.
+In particular, the result's `channels` will not alias with the original.
+"""
+function Base.copy(s::SamplesInfo)
+    return SamplesInfo(map(v -> Base.applicable(copy, v) ? copy(v) : v, getfield(s, :fields)))
```

Comment:
```suggestion
    return SamplesInfo(map(v -> applicable(copy, v) ? copy(v) : v, getfield(s, :fields)))
```
`applicable` is exported from Base.